test/absinthe/schema/type_system_directive_test.exs

```diff
@@ -91,8 +91,137 @@ defmodule Absinthe.Schema.TypeSystemDirectiveTest do
     def resolve_type(_), do: false
   end
 
-  test "Render SDL with Type System Directives applied" do
-    assert Absinthe.Schema.to_sdl(TypeSystemDirectivesSchema) ==
-             TypeSystemDirectivesSchema.sdl()
+  defmodule TypeSystemDirectivesMacroSchema do
+    use Absinthe.Schema
+
+    @prototype_schema WithTypeSystemDirective
+
+    query do
+      field :post, :post do
+        directive :feature, name: ":field_definition"
```

Comment:
I am a bit confused about these strings that look like atoms?  are they arbitrary strings and you just happened to use strings that look like atoms or is there some significance to this format?

Reply:
It's arbitrary, in the same test file there's an SDL schema that follows the same format (https://github.com/absinthe-graphql/absinthe/blob/2390500573c08d882143b48bc02f455ae5d0fd86/test/absinthe/schema/type_system_directive_test.exs). The macro-schema is a near copy of that.